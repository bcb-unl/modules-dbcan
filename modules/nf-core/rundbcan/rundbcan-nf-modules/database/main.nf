process RUNDBCAN_DATABASE {
    label 'process_low'

    conda "${moduleDir}/environment.yml"


    container "ghcr.io/bcb-unl/run_dbcan_new:5.0.2"

    output:

    path "dbcan_db", emit: dbcan_db
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def VERSION = '5.0.2'
    """

    run_dbcan database \\
        --db_dir dbcan_db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def VERSION = '5.0.2'

    """
    mkdir  -p dbcan_db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """

}
