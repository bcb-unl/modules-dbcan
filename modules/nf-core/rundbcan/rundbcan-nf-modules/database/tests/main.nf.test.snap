{
    "rundbcan - database - basic": {
        "content": [
            [
                "versions.yml:md5,ac999108e9bdc59e041c9c9454ff9c8e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-03T18:19:54.233931"
    },
    "rundbcan - database - stub": {
        "content": [
            [
                "versions.yml:md5,ac999108e9bdc59e041c9c9454ff9c8e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-03T18:20:15.150221"
    }
}